### PR TITLE
Risp pulse

### PR DESCRIPTION
**Pull request by kaelyndunnell**
Adds: 

- RISP pulse capability (Issue #16)
- BAKE pulse capability (Issue #17)
- ICWC, GDC pulse capability (Issue #17)
- Python scenario framework to generate input scenario .txt files from user inputs (Issue #27)
- Plasma data handling and festim_model source code to automate W monoblock model 
- Additional tests using pytest 

Changes: 
- Refactors code by moving FESTIM mb model (boundary conditions, material parameters, etc.) to src and leaving user to specify how temperature and particle fluxes evolve over time

### CODE REVIEW

File: environment.yml
Comment (RemDelaporteMathurin) [inline environment.yml:1]: Why is this here since we have the pyproject.toml now?

File: pyproject.toml [hunk 1]
```diff
@@ -15,7 +15,7 @@ requires-python = ">=3.9"
 license = { file = "LICENSE" }
 dependencies = [
     "h_transport_materials",
```
Comment (RemDelaporteMathurin) [inline pyproject.toml:17]: can you pin a HTM version?

File: scenario_test.txt
Comment (RemDelaporteMathurin) [inline scenario_test.txt:1]: Do we still need this file?

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]
